### ISSUE TITLE
Add SQL examples to window functions: `nth_value`, etc

### ISSUE THREAD

**Issue by jonathanc-n** [apache/datafusion#13399]
### Is your feature request related to a problem or challenge?

Mentioned in #13395

### Describe the solution you'd like

Add SQL examples to the window functions (`nth_value`, `cumedist`, etc.)  through the dynamically created docs, an example of it can be found here: #13390 

The window function docs are here: https://datafusion.apache.org/user-guide/sql/window_functions.html

The goal is to have SQL examples for each function

**Comment by jonathanc-n** [comment]
@alamb This can also be given a good first issue label.

**Comment by alamb** [comment]
Thanks @jonathanc-n  -- here are some notes to help anyone who wants to help with this:

What I normaly do is fire up `datafusion-cli` and copy/paste the result of the query into the code

So for example, to  test `cume_dist`

https://datafusion.apache.org/user-guide/sql/window_functions.html#cume-dist

I would do

```shell
> SELECT x, cume_dist() OVER (ORDER BY y) as cume_dist FROM VALUES (1,20), (2, 10), (3,30) t(x,y);
+---+--------------------+
| x | cume_dist          |
+---+--------------------+
| 2 | 0.3333333333333333 |
| 1 | 0.6666666666666666 |
| 3 | 1.0                |
+---+--------------------+
3 row(s) fetched.
Elapsed 0.002 seconds.
```

BTW the syntax 
```sql
VALUES (1,20), (2, 10), (3,30) t(x,y);
```

Basically creates a temporary table named `t` with columns `x` and `y`

**Comment by spencerscott917** [comment]
take

**Comment by rawwar** [comment]
@sageraven1 , Are you still working on this? I see that PR was marked as stale and got closed. If you aren't working on this, I would like to pick this up and complete it.

CC: @alamb

**Comment by alamb** [comment]
> [@sageraven1](https://github.com/sageraven1) , Are you still working on this? I see that PR was marked as stale and got closed. If you aren't working on this, I would like to pick this up and complete it.

That is great! thanks @rawwar 


It appears the functions still do not have examples:
https://datafusion.apache.org/user-guide/sql/window_functions.html#first-value

**Comment by sageraven1** [comment]
> [@sageraven1](https://github.com/sageraven1) , Are you still working on this? I see that PR was marked as stale and got closed. If you aren't working on this, I would like to pick this up and complete it.
> 
> CC: [@alamb](https://github.com/alamb)

Yes, please feel free to take this! Apologies it got stale, life happened.

**Comment by Adez017** [comment]
Hi @alamb  , i am also interested in this ? can I do that ?

**Comment by alamb** [comment]
> Hi [@alamb](https://github.com/alamb) , i am also interested in this ? can I do that ?

Absolutely -- see https://datafusion.apache.org/contributor-guide/index.html#open-contribution-and-assigning-tickets for details on ticket assignments in general
